bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/troubleshoot/StrandDump.java

```diff
@@ -18,59 +18,114 @@
+import java.util.Set;
 
 /**
  * Used to get the status of current Ballerina strands.
  *
  * @since 2201.2.0
  */
 public class StrandDump {
+    private static final String hotSpotBeanName = "com.sun.management:type=HotSpotDiagnostic";
+    private static volatile HotSpotDiagnosticMXBean hotSpotDiagnosticMXBean;
 
     public static String getStrandDump() {
-        Map<Integer, Strand> availableStrands = new HashMap<>();
-        int createdStrandGroupCount = 0;
-        int createdStrandCount = 0;
-        int availableStrandCount = 0;
-        Map<Integer, List<String>> availableStrandGroups = new HashMap<>();
-
-        String strandDumpOutput = generateOutput(availableStrandGroups, availableStrandCount, createdStrandGroupCount,
-                createdStrandCount);
-        cleanUp(availableStrands, availableStrandGroups);
-        return strandDumpOutput;
+        String filename = "thread_dump" + LocalDateTime.now();
+        String dump;
+        try {
+            getStrandDump(System.getProperty("user.dir") + "/" + filename);
+            dump = new String(Files.readAllBytes(Paths.get(filename)));
+            File fileObj = new File(filename);
+            fileObj.delete();
+        } catch (Exception e) {
+            return "Error occurred during strand dump generation";
+        }
+        return generateOutput(dump);
     }
 
-    private static String generateOutput(Map<Integer, List<String>> availableStrandGroups, int availableStrandCount,
-                                         int createdStrandGroupCount, int createdStrandCount) {
+    private static String generateOutput(String dump) {
+        String[] dump_items = dump.split("\\n\\n");
+        int id = 0;
+        Set<Integer> isolatedWorkerList = new HashSet<>();
+        Set<Integer> nonIsolatedWorkerList = new HashSet<>();
+        ArrayList<ArrayList<String>> balTraces = new ArrayList<>();
+        for(String item : dump_items) {
+            String[] lines = item.split("\\n");
+            String[] subitems = lines[0].split("\" ");
+            ArrayList<String> balTraceItems = new ArrayList<>();
+            boolean bal_strand = false;
+            if(subitems.length > 1 && subitems[1].equals("virtual")) {
+                balTraceItems.add("\tStrand " + lines[0].replace("virtual", ":") + "\n\t\tat");
+                String prefix = " ";
+                for(String line : lines) {
+                    if(line.contains("Scheduler.lambda$startIsolatedWorker")) {
+                        isolatedWorkerList.add(id);
+                    }
+                    if(line.contains("Scheduler.lambda$startNonIsolatedWorker")) {
+                        nonIsolatedWorkerList.add(id);
+                    }
+                    if(!line.contains("java") && !line.contains("\" virtual")){
+                        balTraceItems.add(prefix + line + "\n");
+                        prefix = "\t\t   ";
+                    }
+                    if(line.contains(".bal")) {
```

Comment:
We can use constants for strings

Reply:
Fixed with 80fe428001ed625ade093870c89298c84dafa9c3